transactionaldriver-and-tomcat/src/main/java/io/narayana/StringDao.java

```diff
@@ -0,0 +1,102 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
+ */
+public class StringDao {
+
+    private static final String CREATE_TABLE_QUERY = "CREATE TABLE IF NOT EXISTS strings (string VARCHAR(500))";
+
+    private static final String FIND_ALL_STRINGS_QUERY = "SELECT string FROM strings";
+
+    private static final String INSERT_STRING_QUERY = "INSERT INTO strings VALUES ('%s')";
+
+    public StringDao() {
+        initDatabase();
+    }
+
+    /**
+     * Get all strings from the database. This methods closes database connection itself, thus transaction is not needed.
+     * 
+     * @return
+     * @throws SQLException
+     */
+    public List<String> getAll() throws SQLException {
+        List<String> strings = new LinkedList<>();
+        try (Connection connection = getDataSource().getConnection();
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(FIND_ALL_STRINGS_QUERY)) {
+            while (resultSet.next()) {
+                strings.add(resultSet.getString("string"));
+            }
+        }
+        return strings;
+    }
+
+    /**
+     * Save string to the database. This method must be called inside a transaction, because connection is left open to be
+     * closed by transaction manager.
+     *
+     * @param string
+     * @throws SQLException
+     */
+    public void save(String string) throws SQLException {
+        // Connection must be closed by transaction
+        try (Connection connection = getDataSource().getConnection();
+            Statement statement = connection.createStatement()) {
+            statement.execute(String.format(INSERT_STRING_QUERY, string));
+        }
+    }
+
+    /**
+     * Create strings table if it doesn't exist.
+     */
+    private void initDatabase() {
+        try (Connection connection = getDataSource().getConnection(); Statement statement = connection.createStatement()) {
+            statement.execute(CREATE_TABLE_QUERY);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private DataSource getDataSource() {
+        try {
+            return InitialContext.doLookup("java:comp/env/transactionalDataSource");
```

Comment:
IMO it's Tomcat over here: https://github.com/tomjenkinson/quickstart/blob/JBTM-2766/transactionaldriver-and-tomcat/src/main/webapp/META-INF/context.xml#L14